WordPress/WordPressTest/Dashboard/BlogDashboardPersonalizationServiceTests.swift

```diff
@@ -52,4 +52,42 @@ final class BlogDashboardPersonalizationServiceTests: XCTestCase {
             XCTAssertFalse(service.isEnabled(card))
         }
     }
+
+    func testSetEnabledSiteAgnosticReturnsFalseForTheSameSite() {
+        // Given
+        BlogDashboardPersonalizationService(repository: repository, siteID: 1)
+            .setEnabledSiteAgnostic(false, for: .googleDomains)
+
+        // When new service is created
+        let service = BlogDashboardPersonalizationService(repository: repository, siteID: 1)
+
+        // Then settings are retained
+        XCTAssertFalse(service.isEnabled(.googleDomains))
+    }
```

Comment:
❓ I'm curious what's the reason behind the redundant `BlogDashboardPersonalizationService` initialization. The goal is to test the `setEnabledSiteAgnostic` method, here is another way to achieve the same thing:

```swift
func testSetEnabledSiteAgnosticReturnsFalseForTheSameSite() {
    // Given
    let service = BlogDashboardPersonalizationService(repository: repository, siteID: 1)

    // When
    service.setEnabledSiteAgnostic(false, for: .googleDomains)

    // Then
    XCTAssertFalse(service.isEnabled(.googleDomains))
}
```

Reply:
I think the `setEnabledSiteAgnostic` call should be under the `When` section. I asked ChatGPT what the `Given / When / Then` structure means, and here is what it said:

![CleanShot 2023-08-15 at 10 46 46](https://github.com/wordpress-mobile/WordPress-iOS/assets/9609223/d1214d4b-7f3c-4083-9bcb-9af9d6b045ef)